biggraphite/tracing.py

```diff
@@ -16,19 +16,48 @@
 
 try:
     from opencensus.trace import execution_context
+    from opencensus.trace.samplers import always_off
+    from opencensus.trace.tracers import noop_tracer
+    from opencensus.trace import trace_options
+    from opencensus.trace.span_context import SpanContext
 except ImportError:
     execution_context = None
 
 
+def get_bg_trace(self, func):
+    """Prepare the module_name and return a tracer."""
+    if not hasattr(self, 'module_name'):
+        self.module_name = func.__module__.split('.')[-1]
+    return execution_context.get_opencensus_tracer()
+
+
 def trace(func):
     """Decorator for tracing of functions."""
     if not execution_context:
         return func
 
     def tracer(self, *args, **kwargs):
-        if not hasattr(self, 'module_name'):
-            self.module_name = func.__module__.split('.')[-1]
-        _tracer = execution_context.get_opencensus_tracer()
+        _tracer = get_bg_trace(self, func)
         with _tracer.span(name="%s.%s" % (self.module_name, func.__name__)):
             return func(self, *args, **kwargs)
     return tracer
+
+
+def stop_trace():
+    """Stop the current trace."""
+    if not execution_context:
+        pass
+    execution_context.set_current_span(None)
+    tracer = execution_context.get_opencensus_tracer()
+    tracer.tracer = noop_tracer.NoopTracer()
+    tracer.span_context = SpanContext(trace_options=trace_options.TraceOptions(0))
+    tracer.sampler = always_off.AlwaysOffSampler()
```

Comment:
How thread-safe is this block?

What happens if a thread keeps on using the tracer between lines 52 and 54?

Reply:
No thread should share this at this point. To propagate the tracer to another thread it goes through serialization so it will be a copy. If it stills emit a span (which shouldn't happen at this point) it will show as "no root span" on the zipkin/jaeger ui.